sklearn/tests/test_impute.py

```diff
@@ -783,21 +810,49 @@ def test_iterative_imputer_additive_matrix():
     X_test_filled = X_filled[n:]
     X_test = X_missing[n:]
 
-    imputer = IterativeImputer(n_iter=10,
-                               verbose=2,
+    imputer = IterativeImputer(max_iter=10,
+                               verbose=1,
                                random_state=rng).fit(X_train)
     X_test_est = imputer.transform(X_test)
-    assert_allclose(X_test_filled, X_test_est, atol=0.01)
+    assert_allclose(X_test_filled, X_test_est, rtol=1e-3, atol=0.01)
 
 
 def test_iterative_imputer_error_param():
     rng = np.random.RandomState(42)
     X = rng.randn(100, 2)
-    imputer = IterativeImputer(n_iter=-1)
+    imputer = IterativeImputer(max_iter=-1)
     with pytest.raises(ValueError, match='should be a positive integer'):
         imputer.fit_transform(X)
 
 
+def test_iterative_imputer_early_stopping():
+    rng = np.random.RandomState(0)
+    n = 50
+    d = 5
+    max_iter = 100
+    A = rng.rand(n, 1)
+    B = rng.rand(1, d)
+    X = np.dot(A, B)
+    nan_mask = rng.rand(n, d) < 0.5
+    X_missing = X.copy()
+    X_missing[nan_mask] = np.nan
+
+    imputer = IterativeImputer(max_iter=max_iter,
+                               tol=1e-3,
+                               sample_posterior=False,
+                               verbose=1,
+                               random_state=rng)
+    X_filled_100 = imputer.fit_transform(X_missing)
+    assert(len(imputer.imputation_sequence_) == d * 5)
+
+    imputer = IterativeImputer(max_iter=5,
+                               sample_posterior=False,
+                               verbose=1,
+                               random_state=rng)
+    X_filled_5 = imputer.fit_transform(X_missing)
+    assert_allclose(X_filled_100, X_filled_5, atol=1e-7)
```

Comment:
you can add an assert on `n_iter_`. The reason is that we were missing (+1) in some estimator when reporting the number of iteration. it could be quite useful.

Reply:
Sorry, what does this mean? I added this:

```
    imputer = IterativeImputer(max_iter=100,
                               tol=0,
                               sample_posterior=False,
                               verbose=1,
                               random_state=rng)
    imputer.fit(X_missing)
    assert imputer.max_iter == imputer.n_iter_
```

Reply:
Yes, it is what I meant. Inverse the assert order. Usually expected come last.
`assert imputer.n_iter_ == imputer.max_iter`